api/client/client_test.go

```diff
@@ -90,13 +90,13 @@ func TestNewServerGroup(t *testing.T) {
 		t.Fatalf("expected 2 hosts in the group, but have %d", len(hosts))
 	}
 
-	if hosts[0] != "cfssl1.local:8888" {
-		t.Fatalf("expected to see cfssl1.local:8888, but saw %s",
+	if hosts[0] != "http://cfssl1.local:8888" {
+		t.Fatalf("expected to see http://cfssl1.local:8888, but saw %s",
 			hosts[0])
 	}
 
-	if hosts[1] != "cfssl2.local:8888" {
-		t.Fatalf("expected to see cfssl2.local:8888, but saw %s",
+	if hosts[1] != "http://cfssl2.local:8888" {
```

Comment:
You should probably add a tests here with explicit schemes in the `NewServer` call expecting `https://[...]` and `http://[...]`